internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -36,6 +38,26 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
   private class DependencyPhase(prev: Phase) extends GlobalPhase(prev) {
     override def description = "Extracts dependency information"
     def name = Dependency.name
+
+    val executor = Executors.newFixedThreadPool(1)
```

Comment:
This is the simplest possible parallelization - I am open to any more sophisticated (or nicer) ones.
The only problem is that we are in the middle of scala compiler and we've got reduced configuration possibilities.